doc/changelog/07-commands-and-options/10494-diffs-in-show-proof.rst

```diff
@@ -0,0 +1,5 @@
+- Highlight the differences between successive proof steps in the
+  "Show Proof" command if the "Set Diffs" option is enabled.  Experimental;
+  only available in coqtop and Proof General.
```

Comment:
If other IDE maintainers were interested in this, would you be willing to have a try at adding something equivalent to the XML protocol? If yes, I think we should say something like: "may be extended to support other IDEs in the future if there is interest". Otherwise, I really fear that we are sending a message that Proof General gets privileged support despite depending on outdated technology.

Reply:
I tweaked the wording to reflect your comments.

The XML protocol is really not the issue--it already embeds the necessary tags in command output.  The only real issue is about the `Show Proof` code being able to get the previous proof state.  "Experimental" here is a euphemism for "we made a compromise that we will revisit later".

If people later want to change the XML protocol, I'm open to discussion.  Of course, not eager to reinvent things that are working well without good reason (e.g. the diff tags).

With regard to PG, we should understand what's needed to complete the async version and transition the user base to the async version.  Also the benefits from the Coq and PG perspectives that motivate the work.  If the benefits are large, it may make sense to look for a Coq developer to help make it happen, perhaps to do some of the work on the PG side.

Reply:
I tweaked the wording to reflect your comments.

The XML protocol is really not the issue--it already embeds the necessary tags in command output. The only real issue is about the Show Proof code being able to get the previous proof state. "Experimental" here means "we made a compromise that we will revisit later".

If people later want to change the XML protocol, I'm open to discussion. Of course, not eager to reinvent things that are working well without good reason (e.g. the diff tags).

With regard to PG, we should understand what's needed to complete the async version and transition the user base to the async version. Also the benefits from the Coq and PG perspectives that motivate the work. If the benefits are large, it may make sense to look for a Coq developer to help make it happen, perhaps to do some of the work on the PG side.

Reply:
> With regard to PG, we should understand what's needed to complete the async version and transition the user base to the async version. Also the benefits from the Coq and PG perspectives that motivate the work. If the benefits are large, it may make sense to look for a Coq developer to help make it happen, perhaps to do some of the work on the PG side.

I think the path will rather be: first develop new and better IDE protocols (probably LSP since it is what vscode requires), then start to do something about PG. In the meantime, PG works well enough that there is no actual urgency. And there are enough alternative IDEs that test the XML protocol well enough. What we need to ensure is that nobody else writes an IDE based on `coqtop` (as opposed to `coqidetop`), because that's fragile and it creates constraint on the maintenance of Coq (but it seems like it is what is happening already, all other IDEs use `coqidetop` nowadays).

Reply:
OK.

> probably LSP since it is what vscode requires

We should evaluate whether LSP does what we want cleanly rather than adopting it without question.  May be a lot of work to adapt GUIs other than vsCode to use LSP.  Not a simple question.